### PR TITLE
fix some minor issues

### PR DESCRIPTION
**Pull request by YaoZengzeng**
1. extract edsOnly initialization out of loop

2. return if the grpc stream was closed in startPush()

3. return err rather than nil

Signed-off-by: YaoZengzeng <yaozengzeng@zju.edu.cn>

### CODE REVIEW

**Comment by YaoZengzeng** [comment]
@nmittler PTAL :)

File: pilot/pkg/proxy/envoy/v2/ads.go [hunk 1]
```diff
@@ -596,7 +596,7 @@ func (s *DiscoveryServer) StreamAggregatedResources(stream ads.AggregatedDiscove
 
 			err := s.pushConnection(con, pushEv)
 			if err != nil {
-				return nil
+				return err
```
Comment (nmittler) [inline pilot/pkg/proxy/envoy/v2/ads.go]: @costinm was there a reason we were returning nil previously?

Comment (YaoZengzeng) [reply pilot/pkg/proxy/envoy/v2/ads.go]: The code changed here: https://github.com/istio/istio/commit/d455f807ab5191073a70e0d2973b9280869bf6f7#diff-14f4a8a239bf5ba82053cf6606daace5R496

Maybe just a slip of pen?

It was originally returning error.

**Comment by YaoZengzeng** [comment]
ping @nmittler 

Could this PR move on? :)

File: pilot/pkg/proxy/envoy/v2/ads.go (hunk 1 above)
Comment (nmittler) [reply pilot/pkg/proxy/envoy/v2/ads.go]: ping @costinm

**Pull request closed without merging**